schema_enforcer/instances/file.py

```diff
@@ -62,53 +71,97 @@ def __init__(self, root, filename, matches=None):
+        content = self._get_content()
+        top_level_properties = set(content.keys())
+        return top_level_properties
 
-    def _find_matches_inline(self, content=None):
-        """Find addition matches using the Schema ID decorator comment.
+    def _add_matches_by_decorator(self, content=None):
+        """Add matches which declare schema IDs they should adhere to using a decorator comment.
 
-        Look for a line with # jsonschema: schema_id,schema_id
+        If a line of the form # jsonschema: <schema_id>,<schema_id> is defined in the data file, the
+        schema IDs will be added to the list of schema IDs the data will be checked for adherence to.
 
         Args:
             content (string, optional): Content of the file to analyze. Default to None.
 
         Returns:
-            list(string): List of matches found in the file.
+            set(string): Set of matches (strings of schema_ids) found in the file.
         """
         if not content:
-            content = Path(os.path.join(self.full_path, self.filename)).read_text()
+            content = self._get_content(structured=False)
 
-        matches = []
+        matches = set()
 
         if SCHEMA_TAG in content:
             line_regexp = r"^#.*{0}:\s*(.*)$".format(SCHEMA_TAG)
             match = re.match(line_regexp, content, re.MULTILINE)
             if match:
-                matches = [x.strip() for x in match.group(1).split(",")]
+                matches = {x.strip() for x in match.group(1).split(",")}
 
-        return matches
+        matches = matches - self.matches
+        self.matches.update(matches)
 
-    def get_content(self):
-        """Return the content of the instance file in structured format.
+    def _get_content(self, structured=True):
+        """Return the content of the instance file.
 
         Content returned can be either dict or list depending on the content of the file
 
+        Args:
+            structured (bool): Whether to return structured or unstructured data. Defaultes to true.
+
         Returns:
             dict or list: Content of the instance file.
         """
-        return load_file(os.path.join(self.full_path, self.filename))
+        file_location = os.path.join(self.full_path, self.filename)
+
+        if not structured:
+            return Path(file_location).read_text()
+
+        return load_file(file_location)
+
+    def add_matches_by_property_automap(self, schema_manager):
+        """Adds schema_ids to self.matches by automapping top level schema properties to top level keys in instance data.
+
+        Args:
+            schema_manager (schema_enforcer.schemas.manager.SchemaManager): Schema manager oject
+        """
+        matches = set()
+
+        for schema_id, schema_obj in schema_manager.iter_schemas():
+            for top_level_property in schema_obj.top_level_properties:
+                if top_level_property in self.top_level_properties and schema_id not in matches:
```

Comment:
matches is a set, so the `not in matches` is not needed:

```suggestion
                if top_level_property in self.top_level_properties:
```